### DISCUSSION TITLE
I'm having problems generating a C# WPF project

### DISCUSSION THREAD

**Discussion by Igor-rd-Costa** [premake/premake-core#2146] [Q&A]
Hello, I have a Visual Studio solution that contains some C++ project and a WPF project and I'm facing some issues when trying to add the WPF project to the premake setup. Bellow are the relevant solution and project settings.

```
workspace "Orchis"
    configurations { "Debug", "Release", "Dist" }
    architecture "x64"

project "OrchisEditor"
    kind "WindowedApp"
    language "C#"
    location "OrchisEditor"
    dotnetframework "net6.0-windows"
    flags { "WPF" }

    files 
    {
        "OrchisEditor/*.cs",
        "OrchisEditor/View/**.cs",
        "OrchisEditor/Controller/**.cs",
    }

    links
    {
        "OrchisRuntime"
    }
```
I'm having two problems:
1)  The project is not being generated with the correct configurations (it only has `Debug` and `Release` available) and it is not being generated with the correct architecture, C++ projects are being created with a `x64` platform, while the C# project is created with `Any CPU`. This is causing the C# project to not be set to build by default and even after manually enabling the build Visual Studio is skipping it. To fix this and be able to build it I have to open the configuration manager and create a new `x64` project platform for the C# project. Doing this gives me an error that says that it could not created the `x64` configuration because the solution already had a config with that name (not selectable in the C# project). Even with the error the C# project is able to build afterwards, still using the `Any CPU` config. What do I need to do to fix this?

2) Solving the problem described above allows the project to build successfuly but gives me some warning about the `Nullable` setting on the C# project not being enabled. How can I set this in the premake script?

**Comment by gaetandezeiraud** [comment]
@Igor-rd-Costa Any news on it? Have you find a fix or an automatic workaround?
Because I have the same problem.

**Comment by lolrobbe2** [comment]
@gaetandezeiraud this should fix it:

```lua
function platformsElement(cfg)
    _p(2, '<Platforms>AnyCpu;arm64;x64</Platforms>')
end
--should not be needed
function configurationsElement(cfg)
    _p(2, '<Configurations>debug;distribution;release</Configurations>')
end

premake.override(premake.vstudio.cs2005.elements, "projectProperties", function (oldfn, cfg)
    return table.join(oldfn(cfg), {
        platformsElement,
        configurationsElement,
    })
end)
```

i ran in the same problem this should fix it for now.
WPF flag will be replaced by  [dotnetsdk](https://github.com/premake/premake-core/pull/2289)
